pygfx/resources/_texture.py

```diff
@@ -4,27 +4,37 @@
 
 
 class Texture(Resource):
-    """A base texture wrapper that can be implemented for numpy, ctypes arrays,
-    or any other kind of array.
+    """Container for textures.
+
+    A base texture wrapper that can be implemented for numpy, ctypes arrays, or
+    any other kind of array.
 
     Parameters:
-        data (array, optional): Array data of any type that supports the
-            buffer-protocol, (e.g. a bytes or numpy array). If not given
-            or None, nbytes and nitems must be provided. The data is
-            copied if it's float64 or not contiguous.
-        dim (int): The dimensionality of the array (1, 2 or 3).
-        size (3-tuple): The extent ``(width, height, depth)`` of the array.
-            If not given or None, it is derived from dim and the shape of
-            the data. By creating a 2D array with ``depth > 1``, a view can
-            be created with format 'd2_array' or 'cube'.
-        format (str): the format of texture. By default this is automatically
-            set from the data. This must be a pygfx format specifier, e.g. "3xf4",
-            but can also be a format specific to the render backend if necessary
-            (e.g. from ``wgpu.TextureFormat``).
-        colorspace (str): If this data is used as color, it is interpreted to be
-            in this colorspace. Can be "srgb" or "physical". Default "srgb".
-        generate_mipmaps (bool): Whether to automatically generate mipmaps when uploading to
+        data : array, optional
+            Array data of any type that supports the buffer-protocol, (e.g. a
+            bytes or numpy array). If None, nbytes and nitems must be provided.
+            The data is copied if it's not float32 or not contiguous.
+        dim : int
+            The dimensionality of the array (1, 2 or 3).
+        size : tuple, [3]
+            The extent ``(width, height, depth)`` of the array. If None, it is
+            derived from dim and the shape of the data. By creating a 2D array
+            with ``depth > 1``, a view can be created with format 'd2_array' or
+            'cube'.
+            @almarklein: I don't understand this behavior. Can you elaborate so
+            I can come up with a better docstring?
```

Comment:
@almarklein This is the final discussion point. I'm not exactly sure what the second sentence (about the 2D and cube array means).

Reply:
We are down to one unresolved conversation. Once this one is cleared the PR is ready :)